test-integration/util.js

```diff
@@ -67,18 +64,26 @@ const _ignoreKeys = (obj, keys) => {
                 return value.replace(uuid, uuids[uuid]);
             }, obj[key]);
 
-            return { ...x, [key]: value };
-        } else if (keys.indexOf(key) !== -1) {
-            return { ...x, [key]: `___${key}___` };
+            return addElement(x, value, key);
+        } else if (IGNORED_KEYS.includes(key)) {
+            return addElement(x, `___${key}___`, key);
         }
 
-        return { ...x, [key]: _ignoreKeys(obj[key], keys) };
-    }, {});
+        return addElement(x, _ignoreKeys(obj[key]), key);
+    }, Array.isArray(obj) ? [] : {});
 };
 
+const addElement = (obj, elem, key) => {
```

Comment:
Previously `ignoreKeys` didn't recurse into any config that was part of an array. When I changed it to do that, it transformed the arrays into numerically indexed objects, which made the YAML snapshots look bad. This function allows us to preserve arrays as we recurse.